sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsException.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+import com.azure.core.exception.AzureException;
+
+import java.util.Locale;
+
+/**
+ * General exception for Text Analytics related failures.
+ */
+public class TextAnalyticsException extends AzureException {
+    private static final long serialVersionUID = 21436310107606058L;
+
+    private final String errorCodeValue;
+    private final String target;
+
+    /**
+     * Initializes a new instance of the TextAnalyticsException class.
+     * @param message Text containing any additional details of the exception.
+     * @param errorCodeValue The service returned error code value.
+     * @param target The target for this exception.
+     */
+    public TextAnalyticsException(String message, String errorCodeValue, String target) {
+        super(message);
+        this.errorCodeValue = errorCodeValue;
+        this.target = target;
+    }
+
+    @Override
+    public String getMessage() {
+        String baseMessage = super.getMessage();
+
+        if (this.errorCodeValue == null) {
+            return super.getMessage();
+        } else {
+            baseMessage = String.format(Locale.ROOT, "%s %s: {%s}", baseMessage, "ErrorCodeValue",
+                errorCodeValue);
+        }
+
+        if (this.target == null) {
+            return baseMessage;
+        } else {
+            baseMessage = String.format(Locale.ROOT, "%s %s: {%s}", baseMessage, "target", target);
+        }
```

Comment:
Is it possible to have a target without an error code a vice-versa? This seems like it could be cleaned up to check that both aren't 'null' at once and do a full string formatting.

Reply:
It should always have error code but target can be null. Can clean it that way.